src/tests/unit/tests/scanner/custom-rules/aotocomplete-rule.test.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IMock, It, Mock, Times } from 'typemoq';
+import { autocompleteRuleConfiguration } from '../../../../../scanner/custom-rules/autocomplete-rule';
+import { TestDocumentCreator } from './../../../common/test-document-creator';
+
+describe('aotocompleteRule', () => {
+    describe('verify autocomplete rule configs', () => {
+        it('should have correct props', () => {
+            expect(autocompleteRuleConfiguration.rule.id).toBe('autocomplete');
+            expect(autocompleteRuleConfiguration.rule.selector).toEqual(
+                'input[type="text"],\
+input[type="search"],\
+input[type="url"],\
+input[type="tel"],\
+input[type="email"],\
+input[type="password"],\
+input[type="date"],\
+input[type="date-time"],\
+input[type="date-time-local"],\
+input[type="range"],\
+input[type="color"]',
+            );
+            expect(autocompleteRuleConfiguration.rule.any[0]).toBe('autocomplete');
+            expect(autocompleteRuleConfiguration.rule.any.length).toBe(1);
+            expect(autocompleteRuleConfiguration.checks[0].id).toBe('autocomplete');
+        });
+    });
+
+    describe('verify evaluate', () => {
+        let dataSetterMock: IMock<(data) => void>;
+        let testDom: Document;
+
+        beforeEach(() => {
+            dataSetterMock = Mock.ofInstance(data => {});
+        });
+
+        it.each(['text', 'search', 'url', 'tel', 'email', 'password', 'date', 'date-time', 'date-time-local', 'range', 'color'])(
```

Comment:
Should there be some negative test cases in here as well? (eg, verify that it *doesn't* pick up non-input elements with the appropriate attributes, or input elements with inapplicable type attributes?)

Reply:
updated.